shark-android/src/test/java/shark/AndroidReferenceMatcher_XIAMI__RESOURCES_IMPL_Test.kt

```diff
@@ -0,0 +1,173 @@
+
+    val (actualLeakedClassName, actualLeakedRefName) =
+      applicationLeaks.first()
+        .leakTraces.first()
+        .referencePath.first().let {
+      it.owningClassName to it.referenceName
+    }
+
+    assertThat(actualLeakedClassName).isEqualTo(expectedKnownClassLeakingContext)
+    assertThat(actualLeakedRefName).isEqualTo(expectedKnownReferenceName)
+  }
+
+  /**
+   * Validate that this is a known library leak, that we do not report it as an application leak.
+   */
+  private fun HeapAnalysisSuccess.expectKnownLibraryLeak() {
+    // Expect no application leaks found
+    assertThat(applicationLeaks)
+      .describedAs(applicationLeaks.toString()).isEmpty()
+
+    // Expect we see a known library leak instead matching the leak pattern we expect.
+    assertThat(libraryLeaks).hasSize(1)
+    val (actualKnownClassName, actualKnownPattern) =
+      (libraryLeaks.first().pattern as StaticFieldPattern).let {
+        it.className to it.fieldName
+      }
+
+    assertThat(actualKnownClassName).isEqualTo(expectedKnownClassLeakingContext)
+    assertThat(actualKnownPattern).isEqualTo(expectedKnownReferenceName)
+  }
+
+  /**
+   * Sets up the known leak for some manufacturers in a fake, in-memory heap dump.
+   * Then analyzes for leaks with [AndroidReferenceMatchers.XIAMI__RESOURCES_IMPL] added.
+   * The HeapAnalysis is returned for assertions against it.
+   */
+  private fun givenLeakingContextFor(manufacturer: String, sdkInt: Int): HeapAnalysisSuccess {
+
+    val hprofBytes = dumpToBytes {
+      clazz("android.os.Build",
+        staticFields = listOf("MANUFACTURER" to string(manufacturer), "ID" to string("someId"))
+      )
+      clazz("android.os.Build\$VERSION",
+        staticFields = listOf("SDK_INT" to IntHolder(sdkInt))
+      )
+      val leaking = instance(clazz("android.content.Context"))
+      keyedWeakReference(leaking)
+      clazz(
+        expectedKnownClassLeakingContext,
+        staticFields = listOf(expectedKnownReferenceName to leaking)
+      )
+    }
+
+    val matchers = mutableListOf<ReferenceMatcher>()
+    AndroidReferenceMatchers.XIAMI__RESOURCES_IMPL.add(matchers)
+    return hprofBytes.checkForLeaks(referenceMatchers = matchers)
+  }
+
+  fun <T : HeapAnalysis> ByteArray.checkForLeaks(
+    objectInspectors: List<ObjectInspector> = emptyList(),
+    computeRetainedHeapSize: Boolean = false,
+    referenceMatchers: List<ReferenceMatcher> = emptyList(),
+    metadataExtractor: MetadataExtractor = MetadataExtractor.NO_OP,
+    proguardMapping: ProguardMapping? = null,
+    leakingObjectFinder: LeakingObjectFinder = FilteringLeakingObjectFinder(
+      ObjectInspectors.jdkLeakingObjectFilters
+    ),
+  ): T {
+
+    val basp = ByteArraySourceProvider(this)
+
+    val inspectors = if (ObjectInspectors.KEYED_WEAK_REFERENCE !in objectInspectors) {
+      objectInspectors + ObjectInspectors.KEYED_WEAK_REFERENCE
+    } else {
+      objectInspectors
+    }
```

Comment:
Clean this up, you can remove `objectInspectors` as it's always just `listOf(ObjectInspectors.KEYED_WEAK_REFERENCE)`. Same for other params, just remove them as arguments?